scripts/gha/test_simulator.py

```diff
@@ -388,17 +390,61 @@ def _build_tvos_helper(helper_project, device_name, device_os):
 
 
 def _record_apple_tests(video_name):
-  command = "xcrun simctl io booted recordVideo -f --codec=h264 %s" % video_name
-  logging.info("Recording test: %s", command)
-  return subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
-
-
-def _stop_recording(record_process):
-  logging.info("Stop recording test")
-  try:
-    os.killpg(record_process.pid, signal.SIGINT)
-  except:
-    logging.info("Stop recording test failed!!!")
+  return _start_recording([
+    "xcrun",
+    "simctl",
+    "io",
+    "booted",
+    "recordVideo",
+    "-f",
+    "--codec=h264",
+    video_name,
+  ])
+
+
+def _start_recording(args):
+  logging.info("Starting screen recording: %s", subprocess.list2cmdline(args))
+
+  output_file = tempfile.TemporaryFile()
+
+  # Specify CREATE_NEW_PROCESS_GROUP on Windows because it is required for
+  # sending CTRL_C_SIGNAL, which is done by _stop_recording().
+  proc = subprocess.Popen(
+    args,
+    stdout=output_file,
+    stderr=subprocess.STDOUT,
+    creationflags=
+        subprocess.CREATE_NEW_PROCESS_GROUP
+        if platform.system() == "Windows"
+        else 0,
+  )
+  logging.info("Started screen recording with PID %s", proc.pid)
+  return (proc, output_file)
+
+
+def _stop_recording(start_recording_retval):
```

Comment:
note to self: add the logic to upload the captured video to the workflow yml file